src/hugr/hugrmut.rs

```diff
@@ -341,6 +398,37 @@ fn insert_hugr_internal(
     Ok((other_root.into(), node_map))
 }
 
+/// Internal implementation of the `insert_subgraph` method for AsMut<Hugr>.
+///
+/// Returns a mapping from the nodes in the inserted graph to their new indices
+/// in `hugr`.
+///
+/// This function does not update the optypes of the inserted nodes, so the
+/// caller must do that.
+///
+/// In contrast to `insert_hugr_internal`, this function does not preserve
+/// sibling order in the hierarchy.
+fn insert_hugr_internal_with_portgraph(
```

Comment:
I don't understand why this function is being defined instead of using the existing one? I assume in the existing function the graph insertion is done as a side effect, plus returning the inserted root node in the graph seems preferable to putting an `Option` there in the `InsertionResult`

Reply:
`insert_hugr_internal` assumes that you are inserting a complete view. That is, a complete region with a single root node (and maybe more descendants). So we can re-connect the hierarchy of the inserted nodes simply by doing a BFS traversal from the old root.

When inserting a subgraph, we may potentially include nodes from any part of the hugr. We then preserve the hierarchy when both parent and children are being inserted, but if a node's parent is missing then we connect it to the indicated `root`.

Because of that, we cannot do the efficient traversal of `insert_hugr_internal` and instead have to iterate the inserted nodes and ask if their parent is also in the chosen set. That causes the sibling order to not be preserved (unless we add some costly pre-processing).

As for the other point, note the definition of the `InsertionResult`'s `root`:
>     /// The node, after insertion, that was the root of the inserted Hugr.

For a subgraph there is no previous root.

Reply:
Ah I see! Can you add some of this to docstring please? And maybe something like `insert_subgraph` would be a better name since I don't think we can really say "insert hugr" if the hugr we're inserting doesn't have a root